src/memory.h

```diff
@@ -16,28 +16,41 @@ class MemBlock {
   smt::expr writable; // bool::sort()
   smt::expr numelem;  // Index::sort()
 
-  MemBlock(const smt::expr &array, const smt::expr &writable,
-           const smt::expr &numelem):
+  MemBlock(const smt::expr &array, const smt::expr &writable, const smt::expr &numelem):
     array(array), writable(writable), numelem(numelem) {}
 };
 
 class Memory {
 protected:
+  const unsigned int numGlobalBlocks;
+  const unsigned int numLocalBlocks;
```

Comment:
Let's rename this to `maxLocalBlocks`, and rename `currLocalBlocks` to `numLocalBlocks`.